radixdlt-core/radixdlt/src/main/java/com/radixdlt/store/berkeley/BerkeleyLedgerEntryStore.java

```diff
@@ -493,19 +487,62 @@ public void close() {
     return Optional.empty();
   }
 
-  private void storeForkAtEpoch(Transaction dbTxn, long epoch, String forkName) {
-    final var key = new DatabaseEntry(Longs.toByteArray(epoch));
+  private byte[] encodeForkVotingResult(ForkVotingResult forkVotingResult) {
+    return Bytes.concat(
+        FORK_VOTING_RESULT_VERSION_PREFIX,
+        forkVotingResult.candidateForkId().asBytes(),
+        Shorts.toByteArray(forkVotingResult.stakePercentageVoted()));
+  }
+
+  private ForkVotingResult decodeForkVotingResult(long epoch, byte[] data) {
+    final var candidateForkId =
+        HashCode.fromBytes(
+            Arrays.copyOfRange(
+                data,
+                FORK_VOTING_RESULT_VERSION_PREFIX.length,
+                CandidateForkVote.CANDIDATE_FORK_ID_LEN
+                    + FORK_VOTING_RESULT_VERSION_PREFIX.length));
+    final var stakePercentageVoted =
+        Shorts.fromByteArray(
+            Arrays.copyOfRange(
+                data,
+                FORK_VOTING_RESULT_VERSION_PREFIX.length + CandidateForkVote.CANDIDATE_FORK_ID_LEN,
+                data.length));
+    return new ForkVotingResult(epoch, candidateForkId, stakePercentageVoted);
+  }
+
+  private void storeForkAtEpoch(Transaction dbTxn, long newEpoch, String forkName) {
+    final var key = new DatabaseEntry(Longs.toByteArray(newEpoch));
     final var entry = new DatabaseEntry(forkName.getBytes(ForkConfig.FORK_NAME_CHARSET));
     if (forkConfigDatabase.putNoOverwrite(dbTxn, key, entry) != SUCCESS) {
-      throw new BerkeleyStoreException("Duplicate fork hash stored for epoch " + epoch);
+      throw new BerkeleyStoreException("Duplicate fork hash stored for epoch " + newEpoch);
     }
   }
 
   @Override
-  public void storeForkAtEpoch(long epoch, String forkName) {
+  public ImmutableSet<ForkVotingResult> getForksVotingResultsForEpoch(long epoch) {
+    final var result = new ImmutableSet.Builder<ForkVotingResult>();
+    final var value = new DatabaseEntry();
+    try (final var cursor = forksVotingResultsDatabase.openCursor(null, null)) {
+      OperationStatus status =
+          cursor.getSearchKey(new DatabaseEntry(Longs.toByteArray(epoch)), value, DEFAULT);
+      while (status == SUCCESS) {
+        result.add(decodeForkVotingResult(epoch, value.getData()));
+        status = cursor.getNextDup(null, value, DEFAULT);
+      }
+    }
+    return result.build();
+  }
+
+  @Override
+  public void storeForkAtEpoch(long newEpoch, String forkName) {
     final var tx = beginTransaction();
-    storeForkAtEpoch(tx, epoch, forkName);
-    tx.commit();
+    try {
+      storeForkAtEpoch(tx, newEpoch, forkName);
+      tx.commit();
+    } catch (Exception e) {
+      tx.abort();
```

Comment:
The exception is swallowed here - I don't imagine this was intended?

Reply:
Another good catch of yours! This could have potentially left the node in invalid state if left like that.